### PR TITLE
HBW-90 Add payload to widget's requests

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

**Comment by sugarfree1** [comment]
Rename ```paylod``` to ```payload```.

**Comment by mxsmirnoff** [comment]
Updated

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ v1.6.0 [unreleased]
 -   [#207](https://github.com/latera/homs/pull/207) Remove extra locales from moment.js
 -   [#209](https://github.com/latera/homs/pull/209) Drop react-rails & react_ujs dependencies.
 -   [#205](https://github.com/latera/homs/pull/205) Use higher order component instead of HBWSelectMixin.
+-   [#213](https://github.com/latera/homs/pull/213) Add payload to widget's requests
```
Comment (sugarfree1) [inline CHANGELOG.md]: Add dot at the end of line.

File: hbw/app/javascript/packs/hbw/connection.js [hunk 2]
```diff
@@ -216,11 +216,19 @@ modulejs.define('HBWConnection', ['jQuery'], (jQuery) => {
...
       }
 
+      this.payload = options.payload;
+
       this.serverURL = this.options.host + this.options.path;
       this.channels = {};
     }
 
     request (...args) {
+      if (typeof args[0].data === 'undefined') {
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/connection.js]: https://stackoverflow.com/a/3390635
```
args[0].data === undefined
```
is enough. Let's assume that ```undefined``` is not redefined.

**Comment by mxsmirnoff** [comment]
Updated with @TimAle edits

File: hbw/app/javascript/packs/hbw/connection.js [hunk 3]
```diff
@@ -216,12 +216,20 @@ modulejs.define('HBWConnection', ['jQuery'], (jQuery) => {
         this.options.host = `${document.location.protocol}//${document.location.host}`;
       }
 
+      this.payload = options.payload;
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/connection.js]: Hi! ✋ 
You have no reason to do it, payload is already in `this.options.payload` set at line 210.

**Comment by mxsmirnoff** [comment]
Updated

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/connection.js [hunk 4]
```diff
@@ -220,8 +220,19 @@ modulejs.define('HBWConnection', ['jQuery'], (jQuery) => {
       this.channels = {};
     }
 
-    request (...args) {
-      return jQuery.ajax(...args);
+    request (opts) {
+      let data;
+      const payload = this.options.payload || {};
+      console.log(payload);
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/connection.js]: :(

File: hbw/app/javascript/packs/hbw/connection.js [hunk 5]
```diff
@@ -220,8 +220,19 @@ modulejs.define('HBWConnection', ['jQuery'], (jQuery) => {
...
+
+      if (typeof opts.data === 'string') {
+        data = JSON.stringify({ ...JSON.parse(opts.data), payload });
+      } else {
+        data = { ...opts.data, payload };
+      }
+      console.log({ ...opts, data });
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/connection.js]: :(

Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/connection.js]: Sorry

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/connection.js [hunk 6]
```diff
@@ -220,8 +220,17 @@ modulejs.define('HBWConnection', ['jQuery'], (jQuery) => {
       this.channels = {};
     }
 
-    request (...args) {
-      return jQuery.ajax(...args);
+    request (opts) {
+      let data;
+      const payload = this.options.payload || {};
+
+      if (typeof opts.data === 'string') {
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/connection.js]: It is a problem if you're unsure of data type.

**Comment by mxsmirnoff** [comment]
Updated

**Pull request merged**